### PR TITLE
[master < ]Cache user to speed up PullPlan

### PR DESCRIPTION
**Pull request by andrejtonev**
[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

A new FineGrainedAuthChecker is created each time a PullPlan is created. Here we called GetUser(username) which goes to disk and fetches the user data. Caching the user will help speed this up (since the user almost never changes).

### CODE REVIEW

File: src/glue/auth_checker.cpp [hunk 1]
```diff
@@ -94,12 +94,14 @@ std::unique_ptr<memgraph::query::FineGrainedAuthChecker> AuthChecker::GetFineGra
...
     auto locked_auth = auth_->Lock();
-    auto user = locked_auth->GetUser(username);
-    if (!user) {
-      throw memgraph::query::QueryRuntimeException("User '{}' doesn't exist .", username);
+    if (username != user_.username()) {
+      auto maybe_user = locked_auth->GetUser(username);
+      if (!maybe_user) {
+        throw memgraph::query::QueryRuntimeException("User '{}' doesn't exist .", username);
+      }
+      user_ = *maybe_user;
```
Comment (gvolfing) [inline src/glue/auth_checker.cpp:102]:
```suggestion
      user_ = std::move(*maybe_user);
```

**Review by gvolfing — APPROVED** [review]

**Comment by vpavicic** [comment]
@andrejtonev - release note please :)

**Comment by andrejtonev** [comment]
@vpavicic 

Performance improvement when running queries with authorization turned on (with a user).
FineGrainedPermission would read from disk on every query execution. Now we cache the user's info and avoid going to disk until a new user is detected.
NOTE: Changes to the user's privileges will be ignored until next login.

**Comment by vpavicic** [comment]
@andrejtonev ok? 

Users with fine grained privileges will have better performance when running
  queries due to user information cashing. Any changes to the user privileges will
  be ignored until the next login.

**Comment by andrejtonev** [comment]
@vpavicic 
perfect

**Pull request merged**